src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -107,6 +108,8 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
         {
             using (var telemetry = TelemetryActivity.Create(parentId: ParentId, eventName: ProjectRestoreInformation))
             {
+                telemetry.TelemetryEvent.AddPiiData(ProjectFilePath, _request.Project.FilePath);
```

Comment:
We really need to coalesce the 10 events here into 1 :(

Reply:
Created a follow-up issue: https://github.com/NuGet/Client.Engineering/issues/786

Reply:
Oh there's already an existing issue for that :) 
It's just something we haven't scheduled.